### PR TITLE
Add test for Java session state across instances

### PR DESCRIPTION
**Pull request by cdonati**
Reproduces https://github.com/AppScale/appscale/issues/3014.

### CODE REVIEW

File: test-suite/tests/sessions_tests.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    while True:
+      response = self.app.get('/{lang}/sessions/session_handler')
+      self.assertEqual(response.status_code, 200)
+      if response.json()['instanceId'] == instance_id:
+        retry += 1
+        self.assertLess(retry, max_retries)
+        continue
+
+      self.assertEqual(response.json()['name'], name)
+
```
Comment (sjones4) [inline test-suite/tests/sessions_tests.py]: break

File: test-suite/tests/sessions_tests.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+class SessionsTest(HawkeyeTestCase):
+  def test_sessions(self):
+    name = 'user1'
+    response = self.app.post('/{lang}/sessions/session_handler',
+                             data={'name': name})
+    self.assertEqual(response.status_code, 200)
+    instance_id = response.text.strip()
+
```
Comment (sjones4) [inline test-suite/tests/sessions_tests.py]: ```cookies = dict(JSESSIONID=response.cookies['JSESSIONID'])```

grab session cookie.

File: test-suite/tests/sessions_tests.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    # Assuming there are at least two instances for the version
+    # (the appengine-web.xml specifies that there should be at least two), it's
+    # very unlikely for a request to hit the same instance.
+    max_retries = 20
+    retry = 0
+    while True:
+      response = self.app.get('/{lang}/sessions/session_handler')
```
Comment (sjones4) [inline test-suite/tests/sessions_tests.py]:
```
      response = self.app.get('/{lang}/sessions/session_handler',
                              cookies=cookies)
```
Include session cookie in request.

**Comment by cdonati** [comment]
http://ocd.appscale.net:8080/job/Hawkeye/20216/

**Pull request merged**